monai/transforms/croppad/array.py

```diff
@@ -802,16 +804,18 @@ def crop_pad(
         slices = self.compute_slices(roi_start=box_start, roi_end=box_end)
         cropped = super().__call__(img=img, slices=slices)
         pad_to_start = np.maximum(-box_start, 0)
-        pad_to_end = np.maximum(box_end - np.asarray(img.shape[1:]), 0)
+        pad_to_end = np.maximum(
+            box_end - np.asarray(img.peek_pending_shape() if isinstance(img, MetaTensor) else img.shape[1:]), 0
+        )
         pad = list(chain(*zip(pad_to_start.tolist(), pad_to_end.tolist())))
-        pad_width = BorderPad(spatial_border=pad).compute_pad_width(cropped.shape[1:])
+        pad_width = BorderPad(spatial_border=pad).compute_pad_width(
+            cropped.peek_pending_shape() if isinstance(cropped, MetaTensor) else cropped.shape[1:]
+        )
         ret = self.padder.__call__(img=cropped, to_pad=pad_width, mode=mode, **pad_kwargs)
         # combine the traced cropping and padding into one transformation
         # by taking the padded info and placing it in a key inside the crop info.
-        if get_track_meta():
-            ret_: MetaTensor = ret  # type: ignore
-            app_op = ret_.applied_operations.pop(-1)
-            ret_.applied_operations[-1][TraceKeys.EXTRA_INFO]["pad_info"] = app_op
+        if get_track_meta() and isinstance(ret, MetaTensor):
+            ret.applied_operations[-1][TraceKeys.EXTRA_INFO]["pad_info"] = ret.applied_operations.pop()
```

Comment:
Seems when `self.lazy_evaluation=True`, here should be `ret.pending_operations`.

Reply:
after a deep look at it, we don't need to update `pending_operations` here since we need lazy info from `pending_operations`. Here may be updated to
```
if get_track_meta() and isinstance(ret, MetaTensor):
    if not self.lazy_evaluation:
        ret.applied_operations[TraceKeys.EXTRA_INFO]["pad_info"] = ret.applied_operations.pop()
return ret
```

Reply:
Sure, looks good to me